### DISCUSSION TITLE
"cannot bind.   com.github.kr328.clash.service.Tunservice" for android

### DISCUSSION THREAD

**Discussion by Gebarbling** [MetaCubeX/ClashMetaForAndroid#250] [Q&A] [answered]
I am using a Xiaomi phone and have installed the latest version of Clash, but when I start it, a popup shows 'cannot bind.   com.github.kr328.clash.service.Tunservice'. How can I solve this?

**Comment by Gebarbling** [comment, marked as answer]
Restarting the phone solved the problem.

**Reply by jacobeilander** [reply]
It works! Thank you!

**Reply by KaerMalon** [reply]
You are my hero~I have re-downloaded multiple versions and I thought the server crashed

**Comment by Selen-Suyue** [comment]
thank u!!!!!

**Comment by yoyoLooming** [comment]
怎么能完全解决问题呢? 我发现这个问题出现的时候, 手机也无法截图. 小米手机的问题...

**Reply by jacobeilander** [reply]
大家都是用的小米吗,我也是,看来可能是小米的问题

**Reply by xqgg** [reply]
果然小米手机就是一坨屎

**Reply by xqgg** [reply]
应该是系统的大面积BUG，包括无法截图的BUG也已经一年了还没修复

**Reply by jianminYa** [reply]
绷不住了

**Reply by turbodog03** [reply]
三星，同。可能是android的问题？

**Comment by xqgg** [comment]
Don't buy and use Xiaomi phones, it will make you unlucky

**Comment by endsock** [comment]
我也是，每隔几个小时就不行了，重连连接就会报cannot bind. com.github.kr328.clash.service.Tunservice，必须重启手机，每天重启很多次手机，有没有不重启的解决办法？

**Reply by Benxpawa** [reply]
没用

**Comment by rsp4jack** [comment]
可以用 `adb shell dumpsys activity processes` dump 出来 bindService 记录（搜索 `ConnectionRecord`）。
我自己的机器上是 system 有大量对 `com.google.android.gms/.cast.media.CastMediaRoute2ProviderService_Persistent` 的 connection 且全部都是 DEAD 的，可能是 Android 的 bug。

**Reply by endsock** [reply]
怎么解决呢？

**Reply by rsp4jack** [reply]
@endsock 我直接把这个服务禁用掉了。你的设备情况可能不同，建议先 dump 一下看看。
